examples/csrf/samesite/front-envoy.yaml

```diff
@@ -114,3 +114,12 @@ admin:
     socket_address:
       address: 0.0.0.0
       port_value: 8001
+
+layered_runtime:
+  layers:
+  - name: static_layer
+    static_layer:
+      envoy.deprecated_features:envoy.api.v2.route.CorsPolicy.allow_origin: true
+      envoy.deprecated_features:envoy.api.v2.route.Route.per_filter_config: true
+      envoy.deprecated_features:envoy.api.v2.route.VirtualHost.per_filter_config: true
+      envoy.deprecated_features:envoy.type.matcher.StringMatcher.regex: true
```

Comment:
I think these configs are missing from https://github.com/envoyproxy/envoy/blob/master/examples/BUILD. Can we fix the configs? This will have to be updated also https://github.com/envoyproxy/envoy/blob/master/test/config_test/example_configs_test.cc.

Bonus points for figuring out how to automatically make sure new configs get added for testing. This is a consistent problem that we have. Thank you!

Reply:
@mattklein123 i found the problem using this  test runner 

https://github.com/phlax/envoy-examples

weve been talking about how/if we could land it in envoy repo - i was waiting to land this first before thinking  about it

Reply:
OK sounds good. In the interim can you fix the configs and add to the config tests? Thank you.

Reply:
not sure if it would catch these configs tho

Reply:
> I think these configs are missing from https://github.com/envoyproxy/envoy/blob/master/examples/BUILD. Can we fix the configs?

I updated the `BUILD` script with all of the configs with `yaml` suffix.

Not entirely clear if this is correct, but also noticing that there is at least one non-yaml config listed, so there may be more

> This will have to be updated also https://github.com/envoyproxy/envoy/blob/master/test/config_test/example_configs_test.cc.

Im afraid its not immediately obvious to me what needs to change here

Reply:
You need to:
1) Add the missing configs to the build config
2) Fix the configs
3) Update the test to account for the new number of configs (you will see the test will fail).

Thank you!

Reply:
@mattklein123 

i have fixed the deprecation issues in the example configs and updated the necessary test files

I wasnt able to include the `jaeger-native` config as it fails with

```
2020-08-01T17:22:05.6865108Z test/config_test/config_test.cc:134: Failure
2020-08-01T17:22:05.6865445Z Failed
2020-08-01T17:22:05.6866695Z '/b/f/w/_tmp/163cb06a4250265abc95a57e525238d7/test/config_test/jaeger-native-tracing_service1-envoy-jaeger.yaml' config failed. Error: opentracing: failed to load dynamic library: /usr/local/lib/libjaegertracing_plugin.so: cannot open shared object file: No such file or directory
```

im not  sure how to mock/expect the files absence. Im happy to update if i know how.

i also updated the test runner i have tor the examples to test both the `cors` and `csrf`  examples more thoroughly, and ensure i wasnt breaking anything.

regarding keeping the `BUILD` script up-to-date, we could do something like
```
cd examples
configs="$(find . -name "*.yaml" -o -name "*.lua" | grep -v docker-compose | cut  -d/ -f2-)"
for config in $configs; do
    grep "\"$config\"" BUILD || exit 1
done
```
it would fail on this PR because the `jaeger` configs arent   included

Reply:
Awesome thanks a ton. Agreed we can do something like the script you recommend as a follow up? (Perhaps with an exclusion list for configs we know won't work.)